### PR TITLE
Software TPG for the WIB2 format

### PR DESCRIPTION
**Pull request by adam-abed-abud**
Development of the software TPG algorithm for the WIB2 format.

### CODE REVIEW

File: include/fdreadoutlibs/wib2/WIBTPHandler.hpp [hunk 1]
```diff
@@ -0,0 +1,138 @@
+/**
+ * @file WIBTPHandler.hpp Buffer for TPSets
+ *
+ * This is part of the DUNE DAQ , copyright 2021.
+ * Licensing/copyright details are in the COPYING file that you should have
+ * received with this code.
+ */
+#ifndef FDREADOUTLIBS_INCLUDE_FDREADOUTLIBS_WIB_WIBTPHANDLER_HPP_
```
Comment (roland-sipos) [inline include/fdreadoutlibs/wib2/WIBTPHandler.hpp]: Include guard clashing with wib/WIBTPHandler

File: include/fdreadoutlibs/wib2/TPEmulatorModel.hpp [hunk 2]
```diff
@@ -0,0 +1,221 @@
+/**
+ * @file TPEmulatorModel.hpp Emulates a tp source
+ *
+ * This is part of the DUNE DAQ , copyright 2020.
+ * Licensing/copyright details are in the COPYING file that you should have
+ * received with this code.
+ */
+#ifndef FDREADOUTLIBS_INCLUDE_FDREADOUTLIBS_WIB_TPEMULATORMODEL_HPP_
```
Comment (roland-sipos) [inline include/fdreadoutlibs/wib2/TPEmulatorModel.hpp]: Include guard clash.

File: include/fdreadoutlibs/wib2/WIBTPHandler.hpp (hunk 1 above)
Comment (roland-sipos) [reply include/fdreadoutlibs/wib2/WIBTPHandler.hpp]: Maybe also class rename to WIB2TPHandler ?

**Review by roland-sipos — CHANGES_REQUESTED** [review]
See some include guard clashes, and please consider renaming those to WIB2 prefix.

**Comment by roland-sipos** [comment]
Minor problem only with the clashes, otherwise things look good.

**Comment by adam-abed-abud** [comment]
Thank you Roland. I renamed the class to WIB2TPHandler and fixed the guard clash issue.

**Review by eflumerf — APPROVED** [review]
LGTM, compiles, runs.

**Review by roland-sipos — APPROVED** [review]

**Pull request merged**